esrally/track/track.py

```diff
@@ -403,14 +419,15 @@ def __repr__(self):
 
     def __hash__(self):
         return hash(self.name) ^ hash(self.meta_data) ^ hash(self.description) ^ hash(self.challenges) ^ \
-               hash(self.indices) ^ hash(self.data_streams) ^ hash(self.templates) ^ hash(self.corpora)
+               hash(self.indices) ^ hash(self.templates) ^ hash(self.composable_templates) ^ hash(self.component_templates) \
```

Comment:
I have little confidence any of these __hash__ functions actually work but not because of this PR - you can't hash an iterable. Separate PR needed